test/liquity/lqty-staking.js

```diff
@@ -0,0 +1,194 @@
+const { send } = require("@openzeppelin/test-helpers");
+const BigNumber = require("bignumber.js");
+
+const utils = require("../utilities/Utils.js");
+const {
+  depositVault,
+  impersonates,
+  setupCoreProtocol,
+} = require("../utilities/hh-utils.js");
+
+const IERC20 = artifacts.require(
+  "@openzeppelin/contracts/token/ERC20/IERC20.sol:IERC20"
+);
+
+const LiquidatorRegistry = artifacts.require("ILiquidatorRegistry");
+const Strategy = artifacts.require("LQTYStakingStrategyMainnet");
+
+const BorrowerOperations = artifacts.require("IBorrowerOperations");
+const TroveManager = artifacts.require("ITroveManager");
+
+// Vanilla Mocha test. Increased compatibility with tools that integrate Mocha.
+describe("LQTY Staking", function () {
+  let accounts;
+
+  // Parties in the protocol
+  let governance;
+  let farmer;
+  let borrower;
+
+  // Various numbers used in tests
+  let farmerBalance;
+
+  // Core protocol contracts
+  let controller;
+  let vault;
+  let strategy;
+
+  // Underlying LQTY token
+  let underlying;
+
+  // Liquity contracts
+  let borrowerOperations;
+  let troveManager;
+
+  // External setup - use block 12779422
+  let underlyingWhale = "0xF6323aD07c30D696D3e572cb4648814F1188372D";
+  let redeemer = "0x00000000a4569cda123E73F94e35f9a625650Be6";
+
+  const setupExternalContracts = async () => {
+    underlying = await IERC20.at("0x6DEA81C8171D0bA574754EF6F8b412F2Ed88c54D");
+
+    borrowerOperations = await BorrowerOperations.at(
+      "0x24179cd81c9e782a4096035f7ec97fb8b783e007"
+    );
+    troveManager = await TroveManager.at(
+      "0xA39739EF8b0231DbFA0DcdA07d7e29faAbCf4bb2"
+    );
+  };
+
+  const setupBalance = async () => {
+    farmerBalance = await underlying.balanceOf(underlyingWhale);
+
+    // Transfer underlying from whale to farmer
+    await underlying.transfer(farmer, farmerBalance, {
+      from: underlyingWhale,
+    });
+  };
+
+  const setupULPath = async () => {
```

Comment:
The WETH -> LQTY path is available in the UL registry, so it needs to be added before deploying the strategy.

Reply:
Hey, just FYI, the logic for adding liquidation paths is included in `hh-utils.js`. You can use it by adding a `liquidation` argument to the `setupCoreProtocol` function inputs. In this case the line would be:

`liquidation: [{ "uniV3": ["0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2", "0x6DEA81C8171D0bA574754EF6F8b412F2Ed88c54D"]}]`

The liquidation line with the weth-farm path is unnecessary because that path is already stored in the UL. If you would want to add multiple paths in one run it would look like:

`liquidation: [{ "uni": [weth, farm] }, { "uniV3": ["0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2", "0x6DEA81C8171D0bA574754EF6F8b412F2Ed88c54D"]}]`

Reply:
Thanks for the tip!